firestore-bigquery-export/scripts/gen-schema-view/src/index.ts

```diff
@@ -69,141 +82,99 @@ const questions = [
+  if (Object.keys(config.schemas).length === 0) {
     console.log(`No schema files found!`);
   }
   const viewFactory = new FirestoreBigQuerySchemaViewFactory();
-
-  for (const schemaName in schemas) {
-    await viewFactory.initializeSchemaViewResources(datasetId, collectionName, schemaName, schemas[schemaName]);
+  for (let schemaName in config.schemas) {
+    await viewFactory.initializeSchemaViewResources(
+      config.datasetId,
+      config.tableNamePrefix,
+      schemaName,
+      config.schemas[schemaName]
+    );
   }
   return 0;
-};
-
-function readSchemas(globs: string[]): { [schemaName: string]: FirestoreSchema } {
-  let schemas = {};
-  let expanded = expandGlobs(globs);
-  for (var i = 0; i < expanded.length; i++) {
-    let dirent = resolveFilePath(expanded[i]);
-    let stats = lstatSync(dirent);
-    if (stats.isDirectory()) {
-      let directorySchemas = readSchemasFromDirectory(dirent);
-      for (let schemaName in directorySchemas) {
-        if (schemas.hasOwnProperty(schemaName)) {
-          warnDuplicateSchemaName(schemaName);
-        }
-        schemas[schemaName] = directorySchemas[schemaName];
-      }
-    } else {
-      let schemaName = filePathToSchemaName(dirent);
-      if (schemas.hasOwnProperty(schemaName)) {
-        warnDuplicateSchemaName(schemaName);
-      }
-      schemas[schemaName] = readSchemaFromFile(dirent);
-    }
-  }
-  return schemas;
 }
 
-function warnDuplicateSchemaName(schemaName: string) {
-  console.log(`Found multiple schema files named ${schemaName}! Only the latest one will be created!`);
-
-}
-
-function resolveFilePath(filePath: string): string {
-  if (filePath.startsWith(".") || !filePath.startsWith("/")) {
-    filePath = [process.cwd(), filePath].join("/")
-  }
-  return filePath;
-}
-
-function expandGlobs(globs: string[]): string[] {
-  let results = [];
-  for (var i = 0; i < globs.length; i++) {
-    let globResults = glob.sync(globs[i])
-    results = results.concat(globResults);
-  }
-  return results;
-}
-
-function readSchemasFromDirectory(directory: string): { [schemaName: string]: FirestoreSchema } {
-  let results = {};
-  let files = readdirSync(directory);
-  let schemaNames = files.map(fileName => filePathToSchemaName(fileName));
-  for (var i = 0; i < files.length; i++) {
-    const schema: FirestoreSchema = readSchemaFromFile([directory, files[i]].join('/'));
-    results[schemaNames[i]] = schema;
+async function getConfig(): Promise<CliConfig> {
+  let config: CliConfig = {
+    projectId: undefined,
+    datasetId: undefined,
+    tableNamePrefix: undefined,
+    schemas: undefined,
+  };
+  program.parse(process.argv);
+  if (program.nonInteractive) {
+    if (
+      program.project === undefined ||
+      program.dataset === undefined ||
+      program.tableNamePrefix === undefined ||
+      program.schemaFiles.length === 0
+    ) {
+      program.outputHelp();
+      process.exit(1);
+    }
+    config.projectId = program.project;
```

Comment:
[General readability guideline] Avoid setting a variable if you don't have too. Immutability is always preferred.

In a helper method, you can return `CliConfig` from each branch directly without setting.